local_planner/test/test_waypoint_generator.cpp

```diff
@@ -0,0 +1,330 @@
+    avoidance_output.back_off_point.y = 0.6f;
+    avoidance_output.back_off_point.z = 2.2f;
+
+    avoidance_output.back_off_start_point.x = 0.0f;
+    avoidance_output.back_off_start_point.y = 0.0f;
+    avoidance_output.back_off_start_point.z = 2.0f;
+
+    PolarPoint p_pol = histogramIndexToPolar(15, 35, 6, 0.0);
+    avoidance_output.costmap_direction_e = p_pol.e;
+    avoidance_output.costmap_direction_z = p_pol.z;
+
+    avoidance_output.offboard_pose.pose.position.x = 0.0f;
+    avoidance_output.offboard_pose.pose.position.y = 0.0f;
+    avoidance_output.offboard_pose.pose.position.z = 0.0f;
+
+    geometry_msgs::Point n0;
+    n0.x = 0.0f;
+    n0.y = 0.0f;
+    n0.z = 2.0f;
+    geometry_msgs::Point n1;
+    n1.x = 0.8f;
+    n1.y = sqrtf(1 - (n1.x * n1.x));
+    n1.z = 2.0f;
+    geometry_msgs::Point n2;
+    n2.x = 1.3f;
+    n2.y = n1.y + sqrtf(1 - powf(n2.x - n1.x, 2));
+    n2.z = 2.0f;
+    geometry_msgs::Point n3;
+    n3.x = 2.2f;
+    n3.y = n2.y + sqrtf(1 - powf(n3.x - n2.x, 2));
+    n3.z = 2.0f;
+    geometry_msgs::Point n4;
+    n4.x = 2.9f;
+    n4.y = n3.y + sqrtf(1 - powf(n4.x - n3.x, 2));
+    n4.z = 2.0f;
+    geometry_msgs::Point n5;
+    n5.x = 3.5f;
+    n5.y = n4.y + sqrtf(1 - powf(n5.x - n4.x, 2));
+    n5.z = 2.0f;
+    avoidance_output.path_node_positions = {n5, n4, n3, n2, n1, n0};
+
+    position.pose.position.x = 0.0f;
+    position.pose.position.y = 0.0f;
+    position.pose.position.z = 2.0f;
+    position.pose.orientation.x = 0.0f;
+    position.pose.orientation.y = 0.0f;
+    position.pose.orientation.z = 0.0f;
+    position.pose.orientation.w = 1.0f;
+    avoidance_output.pose = position;
+
+    goal.pose.position.x = 10.0f;
+    goal.pose.position.y = 3.0f;
+    goal.pose.position.z = 2.0f;
+
+    goal.pose.orientation.x = 0.0f;
+    goal.pose.orientation.y = 0.0f;
+    goal.pose.orientation.z = 0.0f;
+    goal.pose.orientation.w = 1.0f;
+
+    velocity.twist.linear.x = 2.5f;
+    velocity.twist.linear.y = 3.0f;
+    velocity.twist.linear.z = 0.0f;
+
+    velocity.twist.angular.x = 0.0f;
+    velocity.twist.angular.y = 0.0f;
+    velocity.twist.angular.z = 0.0f;
+
+    updateState(position, goal, velocity, stay, time);
+    setPlannerInfo(avoidance_output);
+    setFOV(270.0, 45.0);
+
+    param_.goal_acceptance_radius_in = 0.5f;
+    param_.goal_acceptance_radius_out = 1.5f;
+    param_.factor_close_to_goal_start_speed_limitation = 3.0f;
+    param_.factor_close_to_goal_stop_speed_limitation = 4.0f;
+    param_.max_speed_close_to_goal_factor = 0.1f;
+    param_.min_speed_close_to_goal = 0.5f;
+  }
+  void TearDown() override {}
+};
+
+TEST_F(WaypointGeneratorTests, goFastTest) {
+  // GIVEN: a waypoint of type goFast
+
+  // WHEN: we generate waypoints
+  waypointResult result = getWaypoints();
+  result = getWaypoints();
+
+  // THEN: we expect a waypoint in between the vehicle curent position and the
+  // goal
+  EXPECT_FLOAT_EQ(0.9578262852f, result.goto_position.x);
```

Comment:
The values you give here are super precise. It is better to give less precise values for the tests unless the API requires those results explicitly (eg calculating a distance, should be precise). Otherwise we end up testing that the implementation is what we implemented, not that the implementation does what the API says it should.